node-runner-cli/api/ValidatorConfig.py

```diff
@@ -108,12 +108,30 @@ def add_change_ownerid(actions: List, validator_info: EntityResponse):
         owner_id = input("\nEnter the new owner id or press Enter not to change:").strip()
         if owner_id != "":
             if owner_id != current_value:
-                actions.append(Action().set_validator_owner(owner_id))
+                actions.append(Action.set_validator_owner(owner_id))
                 return actions
             Helpers.print_coloured_line("Owner entered is same . So action will not be applied", bcolors.WARNING)
 
         return actions
 
+    @staticmethod
+    def vote(actions: List, validator_info: EntityResponse, health, engine_configuration):
+        print("-------- Vote -----\n")
+        latest_fork_name = print_vote_and_fork_info(health, engine_configuration)        
+        response = input(f"{bcolors.BOLD}\nDo you want to cast a vote for fork '{latest_fork_name}' [y/n]? " +
```

Comment:
The more I think about this, the more I'm sure that we should just separate vote casting from registration. Trying to explain to the user here that by voting, they're also registering, feels quite weird / hard to justify?

Why not just have them as separate commands at construction? And then the CLI can make the choice if it wishes to combine them for the user?

What are your thoughts @LukasGasior1  and @Theoklitos ?

Reply:
(Just to check - does Action.vote currently register the validator at the moment? Does registering or updating metadata cause a voting action?)

Reply:
I agree that we should separate vote casting from registration, I will push a change for this

>  does Action.vote currently register the validator at the moment? Does registering or updating metadata cause a voting action?

Yes Action.vote registers the validator. Register _and_ updating metadata cause a voting action. But there is also a third way: Submitting an empty ValidatorSystemMetadata. [Design doc link](https://radixdlt.atlassian.net/wiki/spaces/RPNV1/pages/2606071814/Coordinated+forks+design#How-does-a-validator-change-%2F-remove-their-vote%3F)